src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -248,13 +250,14 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
     val (host, port) = ArmadaUtils.parseMasterUrl(sparkConf.get("spark.master"))
     log(s"host is $host, port is $port")
     val armadaClient = ArmadaClient(host, port)
-    if (armadaClient.submitHealth().isServing) {
+    val healthResp = Await.result(armadaClient.submitHealth(), 5.seconds)
```

Comment:
This should be a configurable duration, with a default, something like this:
https://github.com/armadaproject/armada-spark/blob/master/src/main/scala/org/apache/spark/deploy/armada/Config.scala#L23-L31